<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Greg Harley</author>
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Lithophane Generator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="2" column="0">
     <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,10,0">
      <property name="leftMargin">
       <number>20</number>
      </property>
      <property name="topMargin">
       <number>20</number>
      </property>
      <property name="rightMargin">
       <number>20</number>
      </property>
      <property name="bottomMargin">
       <number>20</number>
      </property>
      <item row="2" column="0">
       <layout class="QVBoxLayout" name="leftSideLayout">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>45</height>
           </size>
          </property>
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Select units for size and heights.  This will convert existing values but model file is always output in millimeters.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="title">
           <string>Units</string>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
          <widget class="QRadioButton" name="btnMetric">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>80</width>
             <height>18</height>
            </rect>
           </property>
           <property name="text">
            <string>millimeters</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
          <widget class="QRadioButton" name="btnImperial">
           <property name="geometry">
            <rect>
             <x>140</x>
             <y>20</y>
             <width>80</width>
             <height>18</height>
            </rect>
           </property>
           <property name="text">
            <string>inches</string>
           </property>
          </widget>
         </widget>
        </item>
        <item>
         <layout class="QFormLayout" name="formLayout">
          <property name="horizontalSpacing">
           <number>0</number>
          </property>
          <property name="verticalSpacing">
           <number>4</number>
          </property>
          <item row="4" column="0">
           <widget class="QLabel" name="label_2">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Maximum height (depth) of generated features. Darkest pixels will have 0 height and lightest will have &lt;span style=&quot; font-style:italic;&quot;&gt;Max Z Height&lt;/span&gt;.  (opposite if &lt;span style=&quot; font-style:italic;&quot;&gt;Invert Image&lt;/span&gt; is selected)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Max Z Height</string>
            </property>
            <property name="buddy">
             <cstring>maxHeight</cstring>
            </property>
            <property name="class" stdset="0">
             <string>input</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="maxHeight">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>5</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_3">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The number of vectors to generate for each pixel.  Higher values generate more detail but increase calculation time.  5 is a good starting point, anything over 10 is probably counter-productive.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Pixels/Vertex</string>
            </property>
            <property name="buddy">
             <cstring>numSamples</cstring>
            </property>
            <property name="class" stdset="0">
             <string>input</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLineEdit" name="numSamples">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>5</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_4">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Height of the base to be added below the generated Lithophane.  This is usually only used if the result is going to be 3D printed.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Base Height</string>
            </property>
            <property name="buddy">
             <cstring>baseHeight</cstring>
            </property>
            <property name="class" stdset="0">
             <string>input</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QLineEdit" name="baseHeight">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_5">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>120</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The minimum or starting height of the generated model.  Usually only necessary if model is to be 3D printed.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Minimum Height</string>
            </property>
            <property name="buddy">
             <cstring>minHeight</cstring>
            </property>
            <property name="class" stdset="0">
             <string>input</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QLineEdit" name="minHeight">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QCheckBox" name="chkInvert">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Create a negative image. i.e. switch light and dark values.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Invert Image</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QCheckBox" name="chkMirror">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Flip the image from left to right about the vertical axis.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Mirror Image</string>
            </property>
           </widget>
          </item>
          <item row="9" column="0" colspan="2">
           <widget class="QCheckBox" name="chkSmooth">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Apply smoothing to the generated result.  Use the slider to specify an amount between &lt;span style=&quot; font-style:italic;&quot;&gt;Min&lt;/span&gt; and &lt;span style=&quot; font-style:italic;&quot;&gt;Max&lt;/span&gt;.  The larger the value the more detail will be lost.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Smoothing</string>
            </property>
           </widget>
          </item>
          <item row="10" column="0" colspan="2">
           <widget class="QSlider" name="sldSmooth">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>20</height>
             </size>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>20</number>
            </property>
            <property name="value">
             <number>10</number>
            </property>
            <property name="tracking">
             <bool>false</bool>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="11" column="0">
           <widget class="QLabel" name="lblMin">
            <property name="text">
             <string>Min</string>
            </property>
           </widget>
          </item>
          <item row="11" column="1">
           <widget class="QLabel" name="lblMax">
            <property name="text">
             <string>Max</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="12" column="0">
           <widget class="QPushButton" name="btnGenerate">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Generate the 3D model.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Generate Lithophane Model</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QGroupBox" name="groupBox_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>50</height>
             </size>
            </property>
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Maximum size of the specified dimension, i.e. width or height as selected.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="title">
             <string>Max Size</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
            <property name="flat">
             <bool>true</bool>
            </property>
            <widget class="QRadioButton" name="btnWidth">
             <property name="geometry">
              <rect>
               <x>120</x>
               <y>20</y>
               <width>80</width>
               <height>18</height>
              </rect>
             </property>
             <property name="text">
              <string>width</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
            <widget class="QRadioButton" name="btnHeight">
             <property name="geometry">
              <rect>
               <x>170</x>
               <y>20</y>
               <width>80</width>
               <height>18</height>
              </rect>
             </property>
             <property name="text">
              <string>height</string>
             </property>
            </widget>
            <widget class="QLabel" name="lblMaxSize">
             <property name="geometry">
              <rect>
               <x>11</x>
               <y>10</y>
               <width>90</width>
               <height>0</height>
              </rect>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>90</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>0</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>Max Size</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
             </property>
             <property name="buddy">
              <cstring>maxSize</cstring>
             </property>
             <property name="class" stdset="0">
              <string>input</string>
             </property>
            </widget>
            <widget class="QLineEdit" name="maxSize">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>20</y>
               <width>100</width>
               <height>22</height>
              </rect>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>100</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>100</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="lblImage">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="2" column="1" colspan="2">
       <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0">
        <property name="sizeConstraint">
         <enum>QLayout::SetNoConstraint</enum>
        </property>
        <item>
         <widget class="QWidget" name="plotWidget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progressBar">
          <property name="value">
           <number>0</number>
          </property>
          <property name="textVisible">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>17</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionLoad_Settings"/>
    <addaction name="actionSave_Settings"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuImage">
    <property name="title">
     <string>Image</string>
    </property>
    <addaction name="actionGenerate"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuImage"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Load Image</string>
   </property>
   <property name="toolTip">
    <string>Load image file</string>
   </property>
   <property name="shortcut">
    <string>Alt+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save 3D Model</string>
   </property>
   <property name="toolTip">
    <string>Export model to .STL file</string>
   </property>
   <property name="shortcut">
    <string>Alt+S</string>
   </property>
  </action>
  <action name="actionGenerate">
   <property name="text">
    <string>Generate</string>
   </property>
   <property name="shortcut">
    <string>Alt+G</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Alt+X</string>
   </property>
  </action>
  <action name="actionLoad_Settings">
   <property name="text">
    <string>Load Settings</string>
   </property>
  </action>
  <action name="actionSave_Settings">
   <property name="text">
    <string>Save Settings</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>btnMetric</tabstop>
  <tabstop>btnImperial</tabstop>
  <tabstop>maxHeight</tabstop>
  <tabstop>numSamples</tabstop>
  <tabstop>baseHeight</tabstop>
  <tabstop>minHeight</tabstop>
  <tabstop>chkInvert</tabstop>
  <tabstop>chkMirror</tabstop>
  <tabstop>chkSmooth</tabstop>
  <tabstop>sldSmooth</tabstop>
  <tabstop>btnGenerate</tabstop>
 </tabstops>
 <resources>
  <include location="litho_gen.qrc"/>
 </resources>
 <connections/>
</ui>
